searx: 1.0.0 -> 1.1.0

nagy opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

nagy committed:
searx: 1.0.0 -> 1.1.0

risicle commented:
@ofborg test searx

Fails for me:

```
uwsgi[825]: --- no python application found, check your startup logs for errors ---
```

nagy commented:
Weird, but I can reproduce it. I was running it with `runInUwsgi = false;` , which works fine. Maybe the uwsgi part in the nixos module needs to be overhauled as well.

dasj19 commented:
for me works with:
```
environment.systemPackages = with pkgs; [
  searx uwsgi
];
```

nagy commented:
> for me works with:
> 
> ```
> environment.systemPackages = with pkgs; [
>   searx uwsgi
> ];
> ```

There is already a `environment.systemPackages = [ cfg.package ];` declaration in the module file, which means the only thing that should be used from your lines is `uwsgi` itself. It hints to me that there are some uwsgi paths that are not pointing to the nix-store, but I cannot find any.

Including `uwsgi` into systemPackages seems wrong to me.

nagy commented:
I have gotten it to work. The problem was, that I have removed the surrounding `toPythonModule` because it did not change the derivation when I removed it and I thought that `buildPythonApplication` already does that. Apparently it messed with uwsgi somehow.

This is now ready for review again.
